core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/set/fields/SetFieldsUsingReflectionUnitTest.java

```diff
@@ -0,0 +1,184 @@
+package com.baeldung.reflection.set.fields;
+
+import java.lang.reflect.Field;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.baeldung.reflection.access.privatefields.Person;
+
+public class SetFieldsUsingReflectionUnitTest {
+
+    @Test
+    public void whenSetIntegerFields_thenSuccess() throws Exception {
+        Person person = new Person();
+
+        Field ageField = person.getClass()
+            .getDeclaredField("age");
+        ageField.setAccessible(true);
+
+        byte age = 26;
+        ageField.setByte(person, age);
+        Assertions.assertEquals(26, person.getAge());
+
+        Field uidNumberField = person.getClass()
+            .getDeclaredField("uidNumber");
+        uidNumberField.setAccessible(true);
+
+        short uidNumber = 5555;
+        uidNumberField.setShort(person, uidNumber);
+        Assertions.assertEquals(5555, person.getUidNumber());
```

Comment:
Change to `Assertions.assertEquals(uidNumber, person.getUidNumber());`. Similarly, change in other places as well